REFERENCE.md

```diff
@@ -0,0 +1,363 @@
+        }
+    }
+}
+```
+
+#### Arguments
+**service**<br>
+A string representing the Learnosity service (API) you want to integrate with. Valid options are:
+
+* assess
+* author
+* data
+* items
+* questions
+* reports
+
+**security**<br>
+An Object that includes your *consumer_key* but does not include your *secret*. The SDK sets defaults for you, but valid options are:
+
+* consumer_key
+* domain (optional)
+* timestamp (optional)
+* user_id (optional)
+
+^Note – the SDK accepts all Objects which can be parsed into a org.json.JSONObject (e.g. a map, a JSONString, JSONOBject or a Java bean).
+
+**secret**<br>
+Your secret key, as provided by Learnosity.
+
+**request**<br>
+An optional associative Object of data relevant to the API being used. This will be any data minus the security details that you would normally use to initialise an API.
+
+^Note – the SDK accepts all Objects which can be parsed into a org.json.JSONObject (e.g. a map, a JSONString, JSONOBject or a Java bean).
+
+
+**action**<br>
+An optional string used only if integrating with the Data API. Valid options are:
+
+* get
+* set
+* update
+* delete
+
+<hr>
+
+### Remote
+
+The Remote class is used to make server side, cross domain requests. Think of it as a cURL wrapper.
+
+You'll call either get() or post() with the following arguments:
+
+* [String] URL
+* [Map<String,Object>]  Data payload
+
+```java
+// Instantiate the SDK Remote class:
+Remote remote = new Remote();
+
+// Instantiate the SDK Remote class with a RequestConfig - https://hc.apache.org/httpcomponents-client-ga/httpclient/apidocs/org/apache/http/client/config/RequestConfig.html:
+RequestConfig requestConfig = RequestConfig.custom().setSocketTimeout(1000).build();
+
+Remote remote = new Remote(requestConfig);
+// Call get() or post() with a URL:
+remote.get("http://schemas.learnosity.com/stable/questions/templates");
+
+// getBody() gives you to body of the request
+String body = response.getBody();
+```
+
+#### Arguments
+
+**URL**<br>
+A string URL, including schema and path. Eg:
+
+```
+https://schemas.learnosity.com/stable/questions/templates
+```
+
+**Data**<br>
+An optional array of data to be sent as a payload. For GET it will be a URL encoded query string.
+
+### Remote methods
+The following methods are available after making a `get()` or `post()`.
+
+**getBody()**<br>
+Returns the body of the response payload.
+
+**getHeader()**<br>
+Currently only returns the *content_type* header of the response.
+
+**getStatusCode()**<br>
+Returns the HTTP status code of the response.
+
+<hr>
+
+### DataApi
+
+This is a helper class for use with the Data API. It creates the initialisation packet and sends a request to the Data API, returning a JSONObject with the response data. There is also a requestRecursive function which can be called with a class implementing RequestCallback.java. The execute() function will be called for each response.
+
+The DataApi Constructor can handle either 3 arguments;
+* url
```

Comment:
Codacy has a fix for the issue: [Lists should be surrounded by blank lines](https://app.codacy.com/gh/Learnosity/learnosity-sdk-java/pullRequest?prid=8556170)
```suggestion

* url
```